JavaScript homeworks/homework-6.js

```diff
@@ -0,0 +1,193 @@
+//Loops task 1//
+let PersonDescription = ['name: John', 'age: 30', 'city: New-York'];
+
+let i = 0;
+let text = '';
+while (PersonDescription[i]) {
+    text += PersonDescription[i] + '\n';
+    i++;
+}
+
+console.log(text);
+
+/*
+name: John
+age: 30
+city: New-York 
+*/
+
+
+//Loops task 2//
+function keyInObject(obj, keyToFind) {
+    for (let key in obj) {
+        if (key == keyToFind) {
+            return true;
+        }
+    }
+    return false;
+}
+
+let person = {
+    name: 'John',
+    age: 30,
+    city: 'New-York'
+};
+
+let keyToFind = 'age';
+if (keyInObject(person, keyToFind)) {
+    console.log('true');
+} else {
+    console.log('false');
+}
+
+// true 
+
+
+//Loops task 3//
+function combineTheWords(obj) {
+    let keys = Object.keys(obj);
+
+    let combineValue = '';
+    let i = 0;
+
+    do {
+        let currentProp = keys[i];
+        combineValue += obj[currentProp];
+        i++;
+
+    } while (i < keys.length);
+
+    return combineValue
+}
+
+let obj = {
+    prop1: 'Hello',
+    prop2: 'World',
+    prop3: '!'
+}
+
+console.log(combineTheWords(obj));
+
+
+//Loops task 4//
+function countNumberOfVowels(string) {
+    let vowels = ['a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U'];
+    let count = 0;
+    let i = 0;
+
+    while (i < string.length) {
+        let char = string[i];
+        if (vowels.includes(char)) {
+            count++;
+        }
+        i++;
+    }
+    return count;
+}
+
+let stringContent = 'Hello world, my name is Albert';
+let result = countNumberOfVowels(stringContent);
+console.log(result); // 8
+
+
+//Loops task 5//
+//v1
+function calculateAverageScore(students) {
+    let average = {};
+
+    students.forEach(s => {
+        let averageScore = s.marks.reduce((a, i) => a + i) / s.marks.length;
```

Comment:
I highly encourage you to use only the tools that you understand. 
If you understand how you got the correct result with reduce, good for you and keep it as it is. 
Bu if it is not clear for you, better use the loops that we already learned

Reply:
+